### DISCUSSION TITLE
How can a default file header be specified ?

### DISCUSSION THREAD

**Discussion by JulesMoorhouse** [SwiftGen/SwiftGen#1014] [Q&A]
I'd like to be able to add a default file header to my project.
This is something I'd normally set in the ...

> MyProject.xcodeproj/project.xcworkspace/xcshareddata/IDETemplateMacros.plist

e.g.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>FILEHEADER</key>
    <string>
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___.
//</string>
</dict>
</plist>
```

Obviously Swiftgen re-creates the Xcodeproj file, so this change is lost.
Is there anyway to add this?
I can't see this in the documentation

Thanks,

jules

**Comment by djbe** [comment]
SwiftGen has nothing to do with your `xcodeproj` file, we don't read it, we don't change it, we don't delete it or (re)create it.

Are you confusing SwiftGen with [XcodeGen](https://github.com/yonaskolb/XcodeGen)?

**Comment by JulesMoorhouse** [comment]
Ah sorry yes, one of those days 👍
